src/fmu/dataio/_model/fields.py

```diff
@@ -404,6 +418,69 @@ class SystemInformation(BaseModel):
     See :class:`OperatingSystem`."""
 
 
+class Tracklog(RootModel):
+    """The ``tracklog`` block contains a record of events recorded on these data.
+    This data object describes the list of tracklog events, in addition to functionality
+    for consturcting a tracklog and adding new records to it.
+    """
+
+    root: List[TracklogEvent]
+
+    def __getitem__(self, item: int) -> TracklogEvent:
+        return self.root[item]
+
+    def __iter__(
+        self,
+    ) -> Any:
+        # Using ´Any´ as return type here as mypy is having issues
+        # resolving the correct type
+        return iter(self.root)
+
+    @classmethod
+    def initialize_metadata_tracklog(cls) -> Tracklog:
+        """Initialize the tracklog object with a list of TracklogEvents
+        with event type 'created'"""
+        return cls(
+            cls._generate_metadata_tracklog_events(enums.TrackLogEventType.created)
+        )
+
+    def extend(self, event: enums.TrackLogEventType) -> None:
+        """Extend the tracklog with a new tracklog record."""
+        self.root.extend(self._generate_metadata_tracklog_events(event))
+
+    @staticmethod
+    def _generate_metadata_tracklog_events(
```

Comment:
I think here also we could skip the 'metadata' part in the name. 
```suggestion
    def _generate_tracklog_events(
```